app/src/main/java/com/kickstarter/viewmodels/SurveyResponseViewModel.java

```diff
@@ -70,41 +56,22 @@ public ViewModel(final @NonNull Environment environment) {
         .compose(bindToLifecycle())
         .subscribe(this.webViewUrl);
 
-      final Observable<Pair<Request, String>> projectRequestAndSurveyUrl = Observable.zip(
+      final Observable<Pair<Request, String>> projectRequestAndSurveyUrl = Observable.combineLatest(
         this.projectUriRequest,
         surveyWebUrl,
         Pair::create
       );
 
-      final Observable<Project> project = projectRequestAndSurveyUrl
-        .filter(requestAndUrl -> !requestTagUrlIsSurveyUrl(requestAndUrl))
-        .map(PairUtils::first)
-        .map(this::extractProjectParams)
-        .switchMap(this.client::fetchProject)
-        .compose(neverError())
-        .share();
-
       projectRequestAndSurveyUrl
         .filter(this::requestTagUrlIsSurveyUrl)
         .compose(ignoreValues())
         .compose(bindToLifecycle())
         .subscribe(this.showConfirmationDialog);
 
-      project
-        .compose(bindToLifecycle())
-        .subscribe(p -> this.startProjectActivity.onNext(Pair.create(p, RefTag.survey())));
-
       this.goBack = this.okButtonClicked;
     }
 
     /**
-     * Parses a project request for project params.
-     */
-    private @NonNull String extractProjectParams(final @NonNull Request request) {
```

Comment:
it's nice that we can remove this logic here, i missed at first that `startProjectActivity`, with project uri request handlers, is baked into our [webview client]!!

Reply:
love [webview clients]